app/pages/notifications/notification-section.jsx

```diff
@@ -143,13 +144,22 @@ export default class NotificationSection extends Component {
     const { notificationsCounter } = this.context;
     const { user } = this.props;
     const { notifications } = this.state;
-    const relatedNotifications = notifications.filter(
-      notification => notification.source_id === readNotification.source_id
-    );
-    relatedNotifications.push(readNotification);
-    const readRequests = relatedNotifications
-      .map(relatedNotification => relatedNotification.update({ delivered: true }).save());
-    Promise.all(readRequests).then(() => notificationsCounter.update(user));
+    const relatedNotifications = notifications.filter(notification => (
+      notification.delivered === false
+        && (notification.id === readNotification.id
+          || notification.source.discussion_id === readNotification.source.discussion_id)
+    ));
```

Comment:
`notification.id === readNotification.id` captures notifications for moderation, data requests, whereas `notification.source.discussion_id === readNotification.source.discussion_id` captures comments (which includes replies and mentions) related to a clicked comment notification

Reply:
LGTM